cli/presenter/presenter.go

```diff
@@ -9,22 +9,39 @@ import (
 	"github.com/spf13/cobra"
 )
 
+// Print writes to stdout.
 func Print(cmd *cobra.Command, args ...interface{}) {
 	_, _ = fmt.Fprint(cmd.OutOrStdout(), args...)
 }
 
+// Println writes to stdout with a newline.
 func Println(cmd *cobra.Command, args ...interface{}) {
 	_, _ = fmt.Fprintln(cmd.OutOrStdout(), args...)
 }
 
+// Printf writes formatted output to stdout.
 func Printf(cmd *cobra.Command, format string, args ...interface{}) {
 	_, _ = fmt.Fprintf(cmd.OutOrStdout(), format, args...)
 }
 
+// PrintSmartf writes formatted output to stdout for human format,
+// or to stderr for structured formats (json, jsonl, raw).
+// Use this for metadata messages that should not pollute structured output.
+func PrintSmartf(cmd *cobra.Command, format string, args ...interface{}) {
+	opts := GetOutputOptions(cmd)
+	if opts.IsStructuredOutput() {
+		Errorf(cmd, format, args...)
+	} else {
+		Printf(cmd, format, args...)
+	}
+}
```

Comment:
any reason why we are printing to stderr here? on another note, lets reconsider the method name

Reply:
We print the actual data to stdout and it  can be piped to jq, other tools. For informational messages, progress, metadata we print it to stderr. This enables: `dirctl search --output json | jq '.[]' `without metadata pollution.

Reply:
What method name do you suggest? PrintMetadataf, PrintContextf or PrintInfof?